### PR TITLE
ARROW-14599: [Release][Java] Upload .jar to Artifacts

### PR DESCRIPTION
**Pull request by kou**
This doesn't upload to any Maven repository. It's out-of-scope of
this.

### CODE REVIEW

File: dev/release/utils-binary.sh [hunk 1]
```diff
@@ -41,6 +41,7 @@ docker_gpg_ssh() {
       -i "${docker_ssh_key}" \
       -p ${ssh_port} \
       -R "/home/arrow/.gnupg/S.gpg-agent:${gpg_agent_extra_socket}" \
+      -t \
```
Comment (emkornfield) [inline dev/release/utils-binary.sh:44]: what does "-t" to here?

**Comment by emkornfield** [comment]
LGTM asked a question about one thing I don't understand.

File: dev/release/utils-binary.sh (hunk 1 above)
Comment (kou) [reply dev/release/utils-binary.sh:44]: This is for suppressing warnings from `gpg`:

```text
warning: Could not set GPG_TTY to stdin: Inappropriate ioctl for device
```

If we add `-t` here, `ssh` allocates a tty. It suppresses the warnings.

**Comment by kou** [comment]
@emkornfield Thanks for your review.

BTW, do you know how to generate not only `.jar` but also `.pom`? It seems that we need `.pom` to upload our artifacts to https://repository.apache.org/#staging-upload . `mvn install ...` generates only `.jar`...: https://github.com/apache/arrow/blob/master/ci/scripts/java_full_build.sh#L30

**Comment by emkornfield** [comment]
According to google "-DgeneratePom=true" added to the "mvn install" should generate the POM (I don't have much experience there)

**Comment by kou** [comment]
Thanks!
I tried it and found that `*.pom` are generated without it. `*.pom` are generated at `~/.m2/repository/org/apache/arrow/` not `target/`.

**Pull request closed without merging**